plugins/action/generate_cloud_modules.py

```diff
@@ -337,6 +337,134 @@ def gen_documentation(
     return documentation
 
 
+def gen_cisco_intersight_documentation(
+    name: str,
+    description: str,
+    parameters: List,
+    added_ins: Dict,
+    next_version: str,
+    target_dir: str,
+) -> Dict:
+
+    short_description = description.split(". ")[0]
+    documentation = {
+        "author": ["Ansible Cloud Team (@ansible-collections)"],
+        "description": description,
+        "module": name,
+        "options": {
+            "api_private_key": {
+                "description": [
+                    "Filename (absolute path) or string of PEM formatted private key data to be used for Intersight API authentication.",
+                    "If a string is used, Ansible vault should be used to encrypt string data.",
+                    "Ex. ansible-vault encrypt_string --vault-id tme@/Users/dsoper/Documents/vault_password_file -----BEGIN EC PRIVATE KEY-----",
+                    "    <your private key data>",
+                    "    -----END EC PRIVATE KEY-----",
+                    "If not set, the value of the INTERSIGHT_API_PRIVATE_KEY environment variable is used."
+                ],
+                "type": "path",
+                "required": True,
+            },
+            "api_uri": {
+                "description": [
+                    "URI used to access the Intersight API.",
+                    "If not set, the value of the INTERSIGHT_API_URI environment variable is used."
+                ],
+                "type": "str",
+                "default": "https://intersight.com/api/v1"
+            },
+            "api_key_id": {
+                "description": [
+                    "Public API Key ID associated with the private key.",
+                    "If not set, the value of the INTERSIGHT_API_KEY_ID environment variable is used."
+                ],
+                "type": "str",
+                "required": True
+            },
+            "validate_certs": {
+                "description": [
+                    "Boolean control for verifying the api_uri TLS certificate"
+                ],
+                "type": "bool",
+                "default": True
+            },
+            "use_proxy": {
+                "description": [
+                    "If C(no), it will not use a proxy, even if one is defined in an environment variable on the target hosts."
+                ],
+                "type": "bool",
+                "default": True
+            },
+        },
+        "requirements": ["python >= 3.9"],
+        "short_description": short_description,
+        "version_added": next_version,
+    }
+
+    # Note: this series of if block is overcomplicated and should
+    # be refactorized.
+    for parameter in parameters:
+        if parameter["name"] == "action":
+            continue
+        normalized_name = normalize_parameter_name(parameter["name"])
```

Comment:
This is specific to vmware. Please check if it is needed for cisco module

Reply:
Removed unneeded code